setup/building-stratum-deps.md

```diff
@@ -0,0 +1,183 @@
+were downloaded by a previous build, you can override the download stage by
+specifying:
+
+```cmake
+  -DDOWNLOAD=no -DFORCE_PATCH=yes
+```  
+
+when you configure the cmake build.
+
+## 5 Build Options
+
+The CMake build script supports the following configuration options.
+
+| Option | Type | Description |
+| ------ | ---- | ----------- |
+| `CMAKE_INSTALL_PREFIX` | Path | Directory in which dependencies should be installed. |
+| `CXX_STANDARD` | Number | C++ standard (11, 14, 17, etc.) the compiler should apply. (Default: not specified) |
+| `DOWNLOAD` | Boolean | Whether to download the source repositories. (Default: TRUE)
+| `FORCE_PATCH` | Boolean | Whether to specify the force (`-f`) option when patching a downloaded repository. (Default: FALSE) |
+| `ON_DEMAND` | Boolean | Whether to build only the specified target(s). If this option is FALSE, all targets will be built. (Default: FALSE) |
+| `USE_LDCONFIG` | Boolean | Whether to use `ldconfig` to update the loader cache after installing a module. Only valid if `USE_SUDO` is enabled. (Default: FALSE) |
+| `USE_SUDO` | Boolean | Whether to use `sudo` to install each module. (Default: FALSE) |
+
+Boolean values are (`TRUE`, `YES`, `ON`) and (`FALSE`, `NO`, `OFF`).
+They may be upper or lower case.
+
+## 6 Examples
+
+### 6.1 Non-root build
+
+To build and install as a nonprivileged user:
+
+```bash
+cd setup
+cmake -B build -DCMAKE_INSTALL_PREFIX=./install
+cmake --build build -j8
+```
+
+The source files will be downloaded, built, and installed in the `setup/install`
+directory. The targets will be built in parallel, using eight threads.
+
+CMake will generate all of its temporary files in the `setup/build` directory.
+This is called an "out-of-source" build.
+
+### 6.2 Non-root build to system directory
+
+To build as a nonprivileged user and use `sudo` to install the libraries to
+a system directory:
+
+```bash
+cmake -B build -DCMAKE_INSTALL_PREFIX=/opt/deps -DUSE_SUDO=yes
+cmake --build build -j6
+```
+
+CMake will build the dependencies as the current user and use `sudo` to
+install the libraries in `/opt/deps`. The build will be done in parallel,
+using six threads.
+
+### 6.3 Root build from source
+
+To build from a source distribution or from repositories that were previously
+downloaded:
+
+```bash
+cmake -B build -DDOWNLOAD=no -DFORCE_PATCH=yes \
+      -DCMAKE_INSTALL_PREFIX=/opt/ipdk/hostdeps
+cmake --build build -j8
+```
+
+CMake will build and install the dependency libraries without downloading the
+source code.
+
+The `FORCE_PATCH` option instructs the patch step to continue without prompting
+if a module has already patched.
+
+### 6.4 On-demand build
+
+To build and install just gRPC and its dependencies:
+
+```bash
+cmake -B build -DON_DEMAND=yes -DCMAKE_INSTALL_PREFIX=~/hostdeps
+cmake --build build -j6 --target grpc
+```
+
+The repositories for grpc and its dependencies will be downloaded, built, and
```

Comment:
`gRPC` instead of `grpc`

Reply:
Actually I wound up rewording the first part of the sentence. It was essentially the same as line 153. That kind of repetition makes for a boring document.